enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,132 @@
+        if not lms_base_url:
+            raise SsoOrchestratorClientError(
+                "Failed to create SSO Orchestrator callback url: LMS_ROOT_URL required.",
+            )
+        path = reverse(
+            'enterprise-customer-sso-configuration-orchestration-complete',
+            kwargs={'configuration_uuid': config_pk},
+        )
+        return urljoin(lms_base_url, path)
+
+    def _get_orchestrator_configure_url(self):
+        """
+        get the configure url for the SSO Orchestrator API
+        """
+        # probably want config value validated for this
+        return urljoin(self.base_url, get_sso_orchestrator_configure_path())
+
+    def _create_auth_header(self):
+        """
+        create the basic auth header for requests to the SSO Orchestrator API
+        """
+        orchestrator_username = get_sso_orchestrator_basic_auth_username()
+        if not orchestrator_username:
+            raise SsoOrchestratorClientError(
+                "Failed to create SSO Orchestrator auth headers: username required.",
+            )
+        orchestrator_password = get_sso_orchestrator_basic_auth_password()
+        if not orchestrator_password:
+            raise SsoOrchestratorClientError(
+                "Failed to create SSO Orchestrator auth headers: password required.",
+            )
+        return HTTPBasicAuth(orchestrator_username, SsoOrchestratorClientError)
+
+    def _create_session(self):
+        """
+        create a requests session object
+        """
+        if not self.session:
+            self.session = requests.Session()
+            self.session.headers['User-Agent'] = USER_AGENT
+            self.session.headers['X-Request-ID'] = get_request_id()
+
+    def _post(self, url, data=None):
+        """
+        make a GET request to the SSO Orchestrator API
+        """
+        self._create_session()
+        response = self.session.post(url, json=data, auth=self._create_auth_header())
+        if response.status_code >= 300:
+            raise SsoOrchestratorClientError(
+                f"Failed to make SSO Orchestrator API request: {response.status_code}",
+                response=response,
+            )
+        return response.status_code
+
+    def configure_sso_orchestration_record(
+        self,
+        config_data,
+        config_pk,
+        enterprise_data,
+        is_sap=False,
+        updating_existing_record=False,
+    ):
+        """
+        configure an SSO orchestration record
+        """
+        callback_url = self._get_orchestrator_callback_url(str(config_pk))
+        if updating_existing_record:
+            callback_url = f"{callback_url}?updating_existing_record=true"
+        config_request_data_key = 'samlConfiguration'
+        if is_sap:
+            config_request_data_key = 'sapsfConfiguration'
+        config_data['uuid'] = str(config_data['uuid'])
+        request_data = {
+            config_request_data_key: config_data,
```

Comment:
We may need both SAML config and SAP SF config for SAP? I thought we still had a metadata URL / XML and such.